sdk/src/main/java/io/dapr/client/DaprClientHttp.java

```diff
@@ -139,12 +139,22 @@ public class DaprClientHttp extends AbstractDaprClient {
    */
   @Override
   public Mono<Void> waitForSidecar(int timeoutInMilliseconds) {
-    return Mono.fromRunnable(() -> {
+    return Mono.defer(() -> {
       try {
+        String[] pathSegments = new String[] { DaprHttp.API_VERSION, "healthz", "outbound"};
+
+        Mono<DaprHttp.Response> responseMono = this.client.invokeApi(DaprHttp.HttpMethods.GET.name(), pathSegments,
+                null, "", null, null);
+        responseMono.block();
+
         NetworkUtils.waitForSocket(Properties.SIDECAR_IP.get(), Properties.HTTP_PORT.get(), timeoutInMilliseconds);
```

Comment:
`NetworkUtils.WaitForSocket(...`

I'm not a Java dev, but It's my understanding that this call should be removed? Otherwise the behaviour hasn't changed? Sorry if I've got this wrong!

Reply:
Also, this check for `healthz/outbound` should probably be put in a loop, and retried on non-2xx until `timeoutInMilliseconds` has elapsed?

Reply:
See the .NET SDK for inspiration :)

https://github.com/dapr/dotnet-sdk/blob/17f849b17505b9a61be1e7bd3e69586718b9fdd3/src/Dapr.Client/DaprClientGrpc.cs#L1758-L1785

Reply:
Agreed this is not needed.

Reply:
I'll rm it and add the retry logic.